contribute/design/1.x/jiva/2019152019-jiva-autosnap-deletion.md

```diff
@@ -0,0 +1,98 @@
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Delete autogenerated snapshot](#delete-autogenerated-snapshot)
+    * [Implementation Details](#implementation-details)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+
+ ## Summary
+
+ This proposal is aimed for the users who experience very frequent network glitch
+ at there cluster, due to which replica get restarted and autogenerated snapshots
+ files are created. A max of 512 snapshots can be created. Once 512 snapshots are
+ created, the replica won't be able to connect again automatically and requires
+ manual cleanup of snapshot.
+
+ ## Motivation
+
+ The motivation behind this approach is to remove/reduce the efforts of the user
+ for managing the storage layer
+
+ ### Goals
+
+ - Ability to remove autogenerated snapshots automatically in background without
+   any manual intervention or with minimal effort.
+
+ ## Proposal
+ Delete the autogenerated snapshots in jiva in the background
+
+ ### User Stories
+ #### Delete autogenerated snapshot:
+ As a user i want to cleanup the autogenerated snapshots with no or minimal efforts
+
+ ### Implementation Details
+
+ Though there is a working flow of deletion of snapshot via REST Api/ jivactl cli
+ but this requires manual intervention from user i.e, user needs to list the snap
+ shots and delete it one by one via given approaches but if number of snapshots have
+ been grown significantly it will take too much time to cleanup those snapshots.
+ There are following approaches which can be used to implement this feature:
+
+ - Delete the n number of snapshots via cli tool jiva
+     * For exp: jivactl purge snapshot -n=10
+ - Delete snapshots having size = 0 or very less amount of data is there ~ 1-5 G
+ - Run a goroutine in controller to perform cleanup if number of snapshot grows to
+   more than n (maybe 10/20...50)
+ - Run a goroutine in replica to perform cleanup of snapshot like above after reload
+   operation.
+
+NOTE: We are going with third approach, where a goroutine will be waiting on a chan,
+once replica get added to controller and verifies the rebuild, snapshot deletion will
+be triggered.
+
+ ## Drawbacks
+
+ - Deletion of snapshot is time taking process since data needs to be merged to
+   its siblings, so larger the data in snapshot, time taken for this will grow
+   significantly and may impact the current IO's.
+ - Third approach and fourth approach is opening the way for many corner cases
+   and not suitable for the cluster where network issue/node maintenance is very
+   frequent, since it's possible that snapshot may not be deleted from one of the
+   replica so the snapshot become stale after the replica restart and require
+   manual cleanup + restart to rebuild again. Although this problem can be mitigated
+   by the deletion of stale snapshots which are not in the chain.
+
+ ## Alternatives
```

Comment:
The above two sections are a little confusing. It is not clear, what is the final approach and if that approach has any limitations. Can the above content be re-structured as follows:

Under the alternatives, mention sub sections for each of the alternate solutions. For example below is for 
(a) Manual Deletion 
(b) Clean up in background

What are the cons of each of the above. 

Then leave the above Implementation Details to the solution that is currently implemented.